app/src/main/java/com/serwylo/lexica/game/Game.java

```diff
@@ -246,8 +252,13 @@ private void initializeDictionary(boolean usDict, boolean ukDict) {
 					ukDict);
 
 			solutions = dict.solver(board,new WordFilter() {
+				//Every word goes through the filter, so do double duty and init maxWordCountsByLength
```

Comment:
I appreciate this type of comment, because I'm generally not a fan of an object such as this `WordFilter` having such strange side effects, but it seems like a perfectly reasonable time to optimize this and prevent another loop later.